### ISSUE TITLE
Handle complex analysis tasks before TCP timeout

### ISSUE THREAD

**Issue by MLHale** [MLHale/funset-enrichment-visualization#1]
I received a feedback comment from a FunSet user indicating they were not receiving timely feedback from the tool when using a large initial dataset (many genes, with many genes in the background set).

We should look at edge cases arising from complex analysis tasks that might be returning large graphs, taking more time than the TCP session timeout.

How should we handle such requests? Maybe an asychronous callback is not sufficient, maybe a websocket is needed in such cases - so that results can be sent back once they are done.

**Comment by thecodingdoc** [comment]
Yes, it’s definitely the size of the dataset. Frankly, I don’t know how he/she can have close to 100,000 annotated genes, when human genes are ~25,000...who knows what they are doing. At any rate, a web socket might be required.
From the user perspective, I’ve often seen an http address being returned for the particular session running, so that users can bookmark it and return to it later when the calculations are done.
As an additional option, we can also invite users to run the enrichment calculations on their machine and then provide the results for visualization to the web server. Of course this would be for more experienced users, and would require (I’m guessing) more extensive tweaking to the server.

On Jul 30, 2019, at 1:28 PM, Matt Hale <notifications@github.com<mailto:notifications@github.com>> wrote:


I received a feedback comment from a FunSet user indicating they were not receiving timely feedback from the tool when using a large initial dataset (many genes, with many genes in the background set).

We should look at edge cases arising from complex analysis tasks that might be returning large graphs, taking more time than the TCP session timeout.

How should we handle such requests? Maybe an asychronous callback is not sufficient, maybe a websocket is needed in such cases - so that results can be sent back once they are done.
